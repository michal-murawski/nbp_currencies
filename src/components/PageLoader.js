import React from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  wrapper: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100hv',
    marginTop: theme.gutters[36]
  }
});

const PageLoader = ({ classes }) => (
  <div className={classes.wrapper}>
    <CircularProgress size={70} thickness={1} />
  </div>
);

PageLoader.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(PageLoader);
