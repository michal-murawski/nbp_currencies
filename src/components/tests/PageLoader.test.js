import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { PageLoaderRaw } from '../PageLoader';

const mockClasses = { wrapper: 'test_value' };

describe('components/PageLoader', () => {
  it('should render properly', () => {
    const wrapper = shallow(<PageLoaderRaw classes={mockClasses} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
