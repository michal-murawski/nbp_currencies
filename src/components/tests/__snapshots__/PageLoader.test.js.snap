// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PageLoader should render properly 1`] = `
<div
  className="test_value"
>
  <WithStyles(CircularProgress)
    size={70}
    thickness={1}
  />
</div>
`;
